command/kv_helpers.go

```diff
@@ -197,3 +199,67 @@ func padEqualSigns(header string, totalLen int) string {
 
 	return fmt.Sprintf("%s %s %s", strings.Repeat("=", equalSigns/2), header, strings.Repeat("=", equalSigns/2))
 }
+
+// listSecretsRecursive dfs-traverses the secrets tree rooted at the given path
+// and returns a list of the leaf paths. Note: for kv-v2, a "metadata" path is
+// expected and "metadata" paths will be returned. If includeDirectories is
+// specified, the output will include non-leaf nodes with "/" suffixes.
+func listSecretsRecursive(ctx context.Context, client *api.Client, path string, includeDirectories bool) ([]string, error) {
+	var descendants []string
+
+	resp, err := client.Logical().ListWithContext(ctx, path)
```

Comment:
this lists _everything_? there's no pagination needed?

Reply:
The result is just a list of keys, it lists all children at the given level.

Reply:
right, but its an exhaustive list?

Reply:
yes, it is